### ISSUE TITLE
Unreproducible segmentation fault bug of ctags

### ISSUE THREAD

**Issue by ssxcxS45XT** [universal-ctags/ctags#2899]
I installed ctags with the following command on macOS.

```
brew tap universal-ctags/universal-ctags
brew install --HEAD universal-ctags
```

Then I git clone the following repo.

```
$ git clone git://sourceware.org/git/glibc.git
```

```
$ ctags --version
Universal Ctags 5.9.0(0071282), Copyright (C) 2015 Universal Ctags Team
Universal Ctags is derived from Exuberant Ctags.
Exuberant Ctags 5.8, Copyright (C) 1996-2009 Darren Hiebert
  Compiled: Mar  3 2021, 19:34:54
  URL: https://ctags.io/
  Optional compiled features: +wildcards, +regex, +iconv, +option-directory, +xpath, +json, +interactive, +yaml, +case-insensitive-filenames, +packcc
$ ctags -R
$ ctags -R
Segmentation fault: 11
```

I got inconsistent results. Sometimes, there is a segmentation fault. Sometimes, there is not. I am not sure what is the problem?

**Comment by masatake** [comment]
What kind of files do you have in the directory where you run ctags.
If possible, I would like to get the archive of the directory.

**Comment by masatake** [comment]
If valgrind is available, run ` valgrind ctags -R`. I would like to see the output.

**Comment by ssxcxS45XT** [comment]
You can download clone the repo and have a test.

**Comment by masatake** [comment]
Which repo are you talking about?

**Comment by masatake** [comment]
Oh, I see. I will try git://sourceware.org/git/glibc.git .

**Comment by masatake** [comment]
Reproduced. 
#2872 is the fix for this issue.
C/C++ maintainer looks busy these days. So the change is not merged yet.

**Comment by masatake** [comment]
I merged #2872. I hope this issue is fixed.
